components/public-api-server/pkg/apiv1/workspace_test.go

```diff
@@ -109,7 +109,7 @@ func TestWorkspaceService_GetWorkspace(t *testing.T) {
 			resp, err := client.GetWorkspace(ctx, &v1.GetWorkspaceRequest{
 				WorkspaceId: scenario.WorkspaceID,
 			})
-			require.Equal(t, scenario.ErrorCode, status.Code(err), "status code must match")
+			require.Equal(t, scenario.ErrorCode.String(), status.Code(err).String(), "status code must match")
 			if diff := cmp.Diff(scenario.Response, resp, protocmp.Transform()); diff != "" {
```

Comment:
Existing could would introduce an `Expectation` type and compare that, rather than use `require`.

Reply:
Unsure if I managed to parse the above correctly. Are you referring to line `if diff := cmp.Diff(scenario.Response, resp, protocmp.Transform()); diff != "" {` or the one above? Github is rendering this against L113.

If this is indeed against line 112, and I understand the above correctly, you're suggesting we do:
```
type Expectation struct {
	Code 			codes.Code
	Response	*v1.GetWorkspaceResponse
}
```
and we would then compare 
```
if diff := cmp.Diff(scenario.Expectation, Expectation{
	Code: status.Code(err),
	Response: resp
})...
```
If this is the case, then the reason for my change to use `.String()` doesn't fully address it. I used `.String()` to get a human readable status code instead of `0x4` or similar hex which is how the status code is represented by grpc.

Would you be able to please clarify what you mean?

Reply:
when using cmp.Diff you'll get a human readable representation of the status code, see https://go.dev/play/p/uv8P_7ZuZ9H

Reply:
Ah, neat. Thanks!

Reply:
Fixed.